forte/data/data_store.py

```diff
@@ -315,50 +397,76 @@ def delete_entry(self, tid: int):
         # We retrieve the entry data from `__entry_dict` using tid. We get its
         # `entry_type`, `type_id`, `begin` and `end` indices. we remove the
         # entry from `__entry_dict` when its information is retrieved. We
-        # find the
-        # `entry_type` sortedlist using `type_id`. We bisect the sortedlist to
-        # find the index of entry data. Then, `type_id` and index are passed to
-        # _delete_entry_by_loc.
+        # find the `entry_type` list using `type_id`. We use `__tid_idx` to
+        # find the index of the entry. Then, the `type_id` and the index are
+        # passed to `_delete_entry_by_loc`.
         raise NotImplementedError
 
     def _delete_entry_by_loc(self, type_id: int, index_id: int):
         r"""It removes an entry of `index_id` by taking both the `type_id`
         and `index_id`. Called by `delete_entry()`.
 
         Args:
-            type_id (int): The index of the sortedlist in `self.__elements`.
-            index_id (int): The index of the entry in the sortedlist.
+            type_id (int): The index of list in `self.__elements`.
+            index_id (int): The index of the entry in the list.
         """
-        # We then remove the entry data from `entry_type` sortedlist.
+        # We then remove the entry data from `entry_type` list.
         raise NotImplementedError
 
     def get_entry(self, tid: int) -> Tuple[List, int, int]:
         r"""This function finds the entry with `tid`. It returns the entry,
-        its `type_id`, and the index in the `entry_type` sortedlist.
+        its `type_id`, and the index in the `entry_type` list.
 
         Args:
             tid (int): Unique id of the entry.
 
         Returns:
             The entry which `tid` corresponds to, its `type_id` and its index
-            in the `entry_type` sortedlist.
-
+            in the `entry_type` list.
         """
+        # If the entry is an annotation, bisect the annotation sortedlist
+        # to find the entry. May use LRU cache to optimize speed.
+        # Otherwise, use `__tid_idx` to find the index of the entry.
         raise NotImplementedError
 
     def get(
         self, entry_type: Union[str, Type[EntryType]], **kwargs
     ) -> Iterator[List]:
-        r"""Fetch entries from the data store of type `entry_type`.
+        r"""This function fetches entries from the data store of
+        type `entry_type`.
 
         Args:
             entry_type: Fully qualified name of this annotation.
 
         Returns:
             An iterator of the entries matching the provided arguments.
-
         """
-        # We find the `type_id` according to `entry_type` and locate the
-        # sortedlist.
-        # We create an iterator to generate entries from the sortedlist.
+        # We find the `type_id` according to `entry_type` and locate the list.
+        # We create an iterator to generate entries from the list.
         yield from self.__elements[self.__type_dict[entry_type]]
```

Comment:
entry_type can be str or EntryType, will this lookup work?

Reply:
Yes I add another function to convert class to str. Do you think it is better to only allow str in the typing?

Reply:
Where is that function called? Here I saw `entry_type` is used directly, which means it should be str.

Reply:
It should be up now. I just pushed it. It is called `_get_entry_type_str`.